### PR TITLE
Port System.Threading documentation from triple slash

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes https://github.com/dotnet/runtime/issues/54866
Used DocsPortingTool to port the remaining triple slash comments.

### CODE REVIEW

File: xml/System.Threading/PreAllocatedOverlapped.xml [hunk 1]
```diff
@@ -222,7 +222,7 @@
         <param name="state">A user-provided object that distinguishes <see cref="T:System.Threading.NativeOverlapped" /> instance produced from this object from other <see cref="T:System.Threading.NativeOverlapped" /> instances. Can be <see langword="null" />.</param>
         <param name="pinData">An object or array of objects representing the input or output buffer for the operations. Each object represents a buffer, for example an array of bytes.  Can be <see langword="null" />.</param>
         <summary>Initializes a new instance of the <see cref="T:System.Threading.PreAllocatedOverlapped" /> class, specifying a delegate that is invoked when each asynchronous I/O operation is complete, a user-provided object providing context, and managed objects that serve as buffers.</summary>
-        <returns>To be added.</returns>
+        <returns>The <see cref="T:System.Threading.PreAllocatedOverlapped" /> instance.</returns>
```
Comment (gewarren) [inline xml/System.Threading/PreAllocatedOverlapped.xml:225]:
```suggestion
        <returns>The new <see cref="T:System.Threading.PreAllocatedOverlapped" /> instance.</returns>
```

File: xml/System.Threading/PeriodicTimer.xml [hunk 2]
```diff
@@ -38,9 +48,11 @@
         <Parameter Name="period" Type="System.TimeSpan" />
       </Parameters>
       <Docs>
-        <param name="period">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="period">The time interval between invocations of the callback.</param>
+        <summary>Initializes the timer.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="period" /> must represent a number of milliseconds larger than 0 and smaller than <see cref="F:System.UInt32.MaxValue" />.</exception>
```
Comment (gewarren) [inline xml/System.Threading/PeriodicTimer.xml:55]:
```suggestion
          <paramref name="period" /> is less than or equal to 0, or greater than <see cref="F:System.UInt32.MaxValue" />.</exception>
```

File: xml/System.Threading/PeriodicTimer.xml [hunk 3]
```diff
@@ -38,9 +48,11 @@
         <Parameter Name="period" Type="System.TimeSpan" />
       </Parameters>
       <Docs>
-        <param name="period">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="period">The time interval between invocations of the callback.</param>
```
Comment (gewarren) [inline xml/System.Threading/PeriodicTimer.xml:51]:
```suggestion
        <param name="period">The time interval in milliseconds between invocations of the callback.</param>
```

**Review by gewarren — APPROVED** [review]

**Comment by adamsitnik** [comment]
@carlossanlop big thanks for adding the missing docs!!

**Pull request merged**